crates/sui-types/src/crypto.rs

```diff
@@ -528,6 +534,59 @@ static_assertions::assert_not_impl_any!(AuthorityWeakQuorumSignInfo: Hash, Eq, P
 impl<const S: bool> AuthoritySignInfoTrait for AuthorityQuorumSignInfo<S> {}
 
 impl<const STRONG_THRESHOLD: bool> AuthorityQuorumSignInfo<STRONG_THRESHOLD> {
+    pub fn new(epoch: EpochId) -> Self {
+        AuthorityQuorumSignInfo {
+            epoch,
+            signatures: vec![],
+            signers_map: RoaringBitmap::new(),
+        }
+    }
+
+    pub fn new_with_signatures(
+        epoch: EpochId,
+        signatures: &Vec<(PublicKeyBytes, AuthoritySignature)>,
+        committee: &Committee,
+    ) -> SuiResult<Self> {
+        let mut sigs = Vec::new();
+        let mut map = RoaringBitmap::new();
+
+        let mut sorted_signatures = signatures.clone();
+        sorted_signatures.sort_by_key(|(a, _)| *a);
+
+        for (pk, sig) in sorted_signatures {
+            sigs.push(sig);
+            map.insert(committee.authority_index(&pk) as u32);
+        }
+
+        Ok(AuthorityQuorumSignInfo {
+            epoch,
+            signatures: sigs,
+            signers_map: map,
+        })
+    }
+
+    // This takes log(sig) time, do not use if necessary
+    pub fn add_signature(
```

Comment:
Here's something that may be useful (food for thought, not for this PR): is there a batch addition of several signatures at the same time, which would perform better?